pkg/instrumentation/metrics.go

```diff
@@ -0,0 +1,42 @@
+package instrumentation
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/collectors"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+// TODO Update metric names according to: https://prometheus.io/docs/instrumenting/writing_exporters/#naming
+const (
+	HttpStatusHistogram = "http_status_histogram"
+)
+
+type Metrics struct {
+	HttpStatusHistogram prometheus.HistogramVec
+
+	reg *prometheus.Registry
+}
+
+// See: https://consoledot.pages.redhat.com/docs/dev/platform-documentation/understanding-slo.html
+// See: https://prometheus.io/docs/tutorials/understanding_metric_types/#types-of-metrics
+func NewMetrics(reg *prometheus.Registry) *Metrics {
+	if reg == nil {
+		panic("reg cannot be nil")
+	}
+	metrics := &Metrics{
+		reg: reg,
+		HttpStatusHistogram: *promauto.With(reg).NewHistogramVec(prometheus.HistogramOpts{
+			Name:    HttpStatusHistogram,
+			Help:    "http status histogram",
+			Buckets: prometheus.LinearBuckets(0.01, 0.05, 10),
```

Comment:
I think you should add `Namespace: "content_sources"` here. This will prepend "content_sources_" to the metric. This will make it easier to identify when using the shared prometheus server.